code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -362,20 +364,20 @@
 /mob/living/simple_animal/hostile/proc/CanSmashTurfs(turf/T)
 	return iswallturf(T) || ismineralturf(T)
 
-/mob/living/simple_animal/hostile/proc/DestroySurroundings()
+/mob/living/simple_animal/hostile/proc/DestroyPathToTarget()
 	if(environment_smash)
 		EscapeConfinement()
-		for(var/dir in GLOB.cardinals)
-			var/turf/T = get_step(targets_from, dir)
-			if(CanSmashTurfs(T))
-				if(T.Adjacent(targets_from))
-					T.attack_animal(src)
-			for(var/a in T)
-				var/atom/A = a
-				if(!A.Adjacent(targets_from))
-					continue
-				if(is_type_in_typecache(A, environment_target_typecache) && !A.IsObscured())
-					A.attack_animal(src)
+		var/turf/T = get_cardinal_step_towards(targets_from, target)
```

Comment:
if the target is diagonal from the mob, it should attack both cardinal directions towards the target at once, to avoid situations where you can position a mob in a spot where it can't attack the stuff it needs to attack to get to you even though it could get to you if it did.

Reply:
i tested this specific circumstance and the mob will attack the target diagonally like normal if the mob is adjacent to the target, and if the target is 1 tile away in a straight diagonal line, the mob will destroy a path like normal

Reply:
![](http://puu.sh/yjgTc/6f4258b16f.png)
The wraith cannot attack the window, since that's not the turf it's finding, and it's unable to break the wall. It needs to be able to attack both relevant turfs for diagonal directions.

Reply:
oh ok gotcha

Reply:
also since it's deterministic that setup in various other orientations may be impossible for mobs to attack through, even if the target is away from it in a straight diagonal line